src/Stryker.Core/Stryker.Core/Compiling/RollbackProcess.cs

```diff
@@ -97,7 +97,11 @@ private SyntaxTree RemoveMutantIfStatements(SyntaxTree originalTree, ICollection
                 {
                     _logger.LogError("Unable to rollback mutation for node {0} with diagnosticmessage {1}", brokenMutation, diagnostic.GetMessage());
                 }
-                brokenMutations.Add(mutationIf);
+
+                if (!brokenMutations.Contains(mutationIf))
```

Comment:
i see that `brokenMutations` is a collection and here `Contains` is called in loop, maybe it should be a HashSet?

Reply:
I'm not sure if that would make much of a difference here. The brokenMutations collection normally contains from 1 to maybe 10 items. If there would be more items this would be a good improvement!